docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+   :start-after: start TUT_get_min_max
+   :end-before: end TUT_get_min_max
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/tutorial/wrapftutorial.f
+   :language: fortran
+   :start-after: start get_min_max
+   :end-before: end get_min_max
+   :dedent: 8
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    call get_min_max(minout, maxout)
+    call assert_equals(-1, minout, "get_min_max minout")
+    call assert_equals(100, maxout, "get_min_max maxout")
+
+Bool
+----
+
+.. ############################################################
+
+.. _example_checkBool:
+
+checkBool
+^^^^^^^^^
+
+Assignments are done to convert between ``logical`` and
+``logical(C_BOOL)``.
+
+C function:
+
+.. literalinclude:: ../regression/run/clibrary/clibrary.c
+   :language: c
+   :start-after: start checkBool
+   :end-before: end checkBool
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: void checkBool(const bool arg1,
+                           bool *arg2+intent(out),
+                           bool *arg3+intent(inout))
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start c_check_bool
+   :end-before: end c_check_bool
+   :dedent: 8
+
+The Fortran wrapper:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start check_bool
+   :end-before: end check_bool
+   :dedent: 4
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    logical rv_logical, wrk_logical
+    rv_logical = .true.
+    wrk_logical = .true.
+    call check_bool(.true., rv_logical, wrk_logical)
+    call assert_false(rv_logical)
+    call assert_false(wrk_logical)
+
+
+Character
+---------
+
+.. ############################################################
+
+.. _example_acceptName:
+
+acceptName
+^^^^^^^^^^
+
+Pass a ``NULL`` terminated string to a C function.
+The string will be unchanged.
+
+.. literalinclude:: ../regression/run/clibrary/clibrary.c
+   :language: c
+   :start-after: start acceptName
+   :end-before: end acceptName
+
+YAML:
+
+.. code-block:: yaml
+
+  - decl: void acceptName(const char *name)
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start c_accept_name
+   :end-before: end c_accept_name
+   :dedent: 8
+
+The Fortran wrapper:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start accept_name
+   :end-before: end accept_name
+   :dedent: 4
+
+No C wrapper is required since the Fortran wrapper calls ``trim`` and
+concatenates ``C_NULL_CHAR``.
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    call accept_name("spot")
+
+.. ############################################################
+
+.. _example_returnOneName:
+
+returnOneName
+^^^^^^^^^^^^^
+
+Pass the pointer to a buffer which the C library will fill.
+The length of the string is unknown.
```

Comment:
Please add a note about where `MAXNAME` comes from in the yaml file.  Especially given the comment that
> the length of the string in unknown

Reply:
Done.  This example is from a user's code. The idea of copying into a char * without knowing the length has lots of issues. But it's probably not uncommon in C code - pre std::string.